test/runtests.jl

```diff
@@ -91,43 +91,52 @@ BinaryProvider.probe_platform_engines!(;verbose=true)
-    @test platform_dlext(:mac64) != platform_dlext(:linuxarmv7l)
+    @test platform_dlext(Linux(:x86_64)) == platform_dlext(Linux(:i686))
+    @test platform_dlext(Windows(:x86_64)) == platform_dlext(Windows(:i686))
+    @test platform_dlext(MacOS()) != platform_dlext(Linux(:armv7l))
 
     # Test some valid dynamic library paths
-    @test valid_dl_path("libfoo.so.1.2.3", :linux64)
-    @test valid_dl_path("libfoo-1.dll", :win64)
-    @test valid_dl_path("libfoo.1.2.3.dylib", :mac64)
-    @test !valid_dl_path("libfoo.dylib", :linux64)
-    @test !valid_dl_path("libfoo.so", :win64)
+    @test valid_dl_path("libfoo.so.1.2.3", Linux(:x86_64))
+    @test valid_dl_path("libfoo-1.dll", Windows(:x86_64))
+    @test valid_dl_path("libfoo.1.2.3.dylib", MacOS())
+    @test !valid_dl_path("libfoo.dylib", Linux(:x86_64))
+    @test !valid_dl_path("libfoo.so", Windows(:x86_64))
 
     # Make sure the platform_key() with explicit triplet works or doesn't
-    @test platform_key("x86_64-linux-gnu") == :linux64
-    @test platform_key("i686-unknown-linux-gnu") == :linux32
-    @test platform_key("x86_64-apple-darwin14") == :mac64
-    @test platform_key("armv7l-pc-linux-gnueabihf") == :linuxarmv7l
-    @test platform_key("aarch64-unknown-linux-gnu") == :linuxaarch64
-    @test platform_key("powerpc64le-linux-gnu") == :linuxppc64le
-    @test platform_key("x86_64-w64-mingw32") == :win64
-    @test platform_key("i686-w64-mingw32") == :win32
-    @test_throws ErrorException platform_key("invalid-triplet-yo")
-    @test_throws ErrorException platform_key("aarch64-unknown-gnueabihf")
-    @test_throws ErrorException platform_key("x86_64-w32-mingw64")
+    @test platform_key("x86_64-linux-gnu") == Linux(:x86_64)
+    @test platform_key("i686-unknown-linux-gnu") == Linux(:i686)
+    @test platform_key("x86_64-apple-darwin14") == MacOS()
+    @test platform_key("armv7l-pc-linux-gnueabihf") == Linux(:armv7l)
+    @test platform_key("aarch64-unknown-linux-gnu") == Linux(:aarch64)
+    @test platform_key("powerpc64le-linux-gnu") == Linux(:ppc64le)
+    @test platform_key("x86_64-w64-mingw32") == Windows(:x86_64)
+    @test platform_key("i686-w64-mingw32") == Windows(:i686)
+    @test_throws ArgumentError platform_key("invalid-triplet-yo")
+    @test_throws ArgumentError platform_key("aarch64-unknown-gnueabihf")
+    @test_throws ArgumentError platform_key("x86_64-w32-mingw64")
 
     # Test that we can indeed ask if something is linux or windows, etc...
-    @test platform_is_linux(:linuxaarch64)
-    @test !platform_is_linux(:win64)
-    @test platform_is_windows(:win32)
-    @test !platform_is_windows(:linux64)
-    @test platform_is_apple(:mac64)
-    @test !platform_is_apple(:linuxppc64le)
+    @test Compat.Sys.islinux(Linux(:aarch64))
+    @test !Compat.Sys.islinux(Windows(:x86_64))
+    @test Compat.Sys.iswindows(Windows(:i686))
+    @test !Compat.Sys.iswindows(Linux(:x86_64))
+    @test Compat.Sys.isapple(MacOS())
+    @test !Compat.Sys.isapple(Linux(:ppc64le))
```

Comment:
You need to add a `using Compat` at the top of this file.  How does this possibly pass Travis on 0.6?!

Reply:
I... don't know.

Reply:
This more than anything convinces me that I have absolutely no idea how computers work.

Reply:
Oddly enough, 0.7 caught it but not 0.6.

Reply:
And AppVeyor didn't seem to notice

Reply:
TwilightZone.jl